Assets/Scripts/Controllers/CharController.cs

```diff
@@ -158,10 +159,11 @@ protected void Rotate(Vector3 direction)
     }
 
 
-    protected bool MeleeAttack(LayerMask mask)
+    protected virtual string[] MeleeAttack(LayerMask mask)
```

Comment:
It seems dangerous to link gold progression to the melee attack or to characters in general. Perhaps a better way to do it would be with a gold manager that increments a value whenever an enemyController is disabled.

With the current system we'd have to copy this logic to the card attacks too, and that can get really messy

Reply:
We should maybe also discuss this on a call and draw up some simple architecture models for this

Reply:
It was the simplest way to trigger on an enemy kill. I don't want it bring Triggered on enemy death. Because if enemies somehow die on their own, or DeShawn or whatever, it'd be incrementing your gold. so I'd rather keep it like this and change how a kill is Triggered.

Reply:
idk why it seems "dangerous" other than the fact that it was just me working around existing code.

Reply:
Honestly man there's better shit I think we should be focusing our time on. Redoing PRs because we disagree on a line of code seems pointless